From: Chris Mason <clm@fb.com>
Date: Sat, 11 Apr 2015 05:09:06 -0700
Patch-mainline: 4.1
Git-commit: 2eb8c62e317b2305da00dc7986a982fa9637fedc
Subject: [PATCH] Btrfs: fix uninit variable in clone ioctl

Commit 0d97a64e0 creates a new variable but doesn't always set it up.
This puts it back to the original method (key.offset + 1) for the cases
not covered by Filipe's new logic.

Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index d79c599240a7..64e8fb639f72 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -3206,7 +3206,7 @@ static int btrfs_clone(struct inode *src, struct inode *inode,
 	key.offset = off;
 
 	while (1) {
-		u64 next_key_min_offset;
+		u64 next_key_min_offset = key.offset + 1;
 
 		/*
 		 * note the key will change type as we walk through the
-- 
2.1.3

